.github/actions/package/action.yml

```diff
@@ -0,0 +1,106 @@
+name: Package
+description: Packages the SDK
+
+inputs:
+  node-version:
+    description: The version of Node.js to use
+    required: true
+  java-version:
+    description: The version of Java to use
+    required: true
+  vtag:
+    description: The version tag to be used for the SDK
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v2
+      with:
+        node-version: ${{ inputs.node-version }}
+        cache: 'npm'
+
+    - name: Use JDK 11
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: ${{ inputs.java-version }}
+
+    - name: Install dependencies
+      if: steps.node-cache.outputs.cache-hit != 'true'
+      run: npm ci
+      shell: bash
+
+    - name: Cache Gradle packages
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.gradle/caches
+          ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: android-build
+
+    - name: Extract Android build output
+      run: tar -xzvf android-build.tar.gz
+      shell: bash
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: ios-build
+
+    - name: Extract iOS build output
+      run: tar -xzvf ios-build.tar.gz
+      shell: bash
+
+    - name: Cache Native Modules
```

Comment:
Well, an `npm run build` wont touch modules because it runs `scons build` which just builds for the requested platforms, it does not package the SDK which is where the module downloads are done (if you ran `npm run cleanbuild` or `npm run package` you would see this happen)

The modules are pulled down within [includePackagedModules](https://github.com/tidev/titanium_mobile/blob/master/build/lib/packager.js#L218), with how the scons scripts are written I guess yes something specific is done as that function is [called for each OS requested](https://github.com/tidev/titanium_mobile/blob/master/build/lib/builder.js#L220-L222)